lib/workers/global/autodiscover.ts

```diff
@@ -30,7 +30,17 @@ export async function autodiscoverRepositories(
     return config;
   }
   if (config.autodiscoverFilter) {
-    discovered = discovered.filter(minimatch.filter(config.autodiscoverFilter));
+    const matched: string[] = [];
+
+    for (const filter of config.autodiscoverFilter) {
+      matched.push(...discovered.filter(minimatch.filter(filter)));
+    }
```

Comment:
```suggestion
  const filters = config.autodiscoverFilter.map(f => minimatch.filter(f));
  for (const repo of discovered) { 
      for (const filter of filters) {
        if (filter(repo)) {
          matched.add(repo);
        }
      }
    }
```

Reply:
Isn't possible, since `filter` has the following signature: `(element: string, indexed: number, array: readonly string[])`.
So there are two parameters missing.

I'm not a full-time JavaScript/TypeScript developer, so could you please add a comment on why those changes need to be made. This would be nice because this way I will be able to learn from my mistakes.

Reply:
My main issue is that the repo list can be very long and your code recreates the Minimatch filter on every iteration.

So the filter creation should be outside of the repo loop.

Also the set will be faster then your index of unique loop.

Reply:
Ok thanks :+1:

How do I address the function signature problem?

Reply:
I think you need an other Minimatch function

https://github.com/isaacs/minimatch#minimatchmatchlist-pattern-options

Reply:
Sorry for my late response.

I've fixed the discussed points.